### ISSUE TITLE
Provide preset storage states that any preset can leverage

### ISSUE THREAD

**Issue by jenlampton** [backdrop-contrib/preset#1]
In backdrop, everything that has a storage state (image styles, views, layouts, etc) needs to define its own constants to represent those states. This means that every conrib project with a preset, would also need to implement it's own set of identical globals.

Until we can get a single state [into core](https://github.com/backdrop/backdrop-issues/issues/1345),
it would be nice if preset could provide this single set of constants to indicate whether a preset is default or overridden. 

Additionally, this state could automatically be displayed on the list form (as opposed to needing to be manually added to the form by each preset). 

If you are open to this idea, I'd be willing to work on a proof of concept.

**Comment by ghost** [comment]
@jenlampton Thanks for the suggestion. Sure, would love to see a POC!

**Comment by ghost** [comment]
At first I wasn't sure how this'd be relevant to this module. But then I realised you can provide default presets in config and then users can edit them themselves. So yes, I see the benefit in having the ability to show default/overridden presets, and the ability to revert overridden ones back to default.

**Comment by jenlampton** [comment]
Yes, exactly. Video Embed field ships with one default "Normal" Video Style so I was looking for a way to indicate if it had been overridden or not.
